synce_port.c

```diff
@@ -280,6 +280,8 @@ void synce_port_destroy(struct synce_port *port)
 	} else {
 		pr_warning("%s pc is NULL", __func__);
 	}
+
+	free(port);
```

Comment:
We actually do release it in here: https://github.com/intel/synce4l/blob/main/synce_dev.c#L51 
Do you think it makes sense that `synce_dev` both create and destroy `ports` or should we consider changing it?

Btw. we do have internal unit tests but due to some issues couldn't release them yet. We do our best to make sure we don't leak any memory and test check that automatically. Either way much thanks for taking care!

Reply:
It would make more sense to me if the destroy functions freed the memory, but whatever you choose please do it consistently.

I'll push a fixed commit that moves the missing frees to the correct place to avoid double free.